react/interfaces.tsx

```diff
@@ -0,0 +1,15 @@
+import { VideoHTMLAttributes } from 'react'
+
+export interface VideoPlayer extends VideoHTMLAttributes<HTMLVideoElement> {
+  name?: string
+  description?: string
+  src: string
+  type?: string
+  controlsType?: string
+  IconPlay?: any
+  IconPause?: any
+  IconFullscreen?: any
+  IconVolumeOn?: any
+  IconVolumeOff?: any
```

Comment:
You could type all components here, gonna make some examples of how and you can choose one

```suggestion
  IconPlay?: React.ComponentType<unknown> // Not the best I think, because it accepts class components
  IconPause?: React.FC<unknown> // Maybe the best way
  IconFullscreen?: typeof IconFullsscreenDefault // I don't like this one (and I think is wrong as well)
  IconVolumeOn?: React.FC<any> // I think typing the props with unknown is better because we don't pass anything to it
  IconVolumeOff?: React.FC<{}> This might be an option too if we think about what I wrote in the above example. This way we are saying to the user that he/she shouldn't expect any props to be passed to the component
```

Reply:
wow @Klynger, thanks!!